### PR TITLE
Send the entity to send fewer bits than sending the class

### PR DESCRIPTION
**Pull request by MoskalykA**
Hi
I suggest sending an entity to send fewer bits than sending the class, which is the same as sending a string.
An entity corresponds to 13 bits whereas a string with two characters corresponds to a higher number of bits and we know that the class of an entity is not two characters long.

### CODE REVIEW

**Comment by robotboy655** [comment]
Have you actually tested this change on an actual server?

**Comment by Kefta** [comment]
I believe net messages will always reach the client before its knowledge of entity deletion when dispatched at the same time, but this should be tested extensively. I'll do some this weekend.

File: garrysmod/gamemodes/base/gamemode/npc.lua [hunk 1]
```diff
@@ -3,6 +3,8 @@
 util.AddNetworkString( "PlayerKilledNPC" )
 util.AddNetworkString( "NPCKilledNPC" )
 
+local worldspawn = Entity(0)
```
Comment (Kefta) [inline garrysmod/gamemodes/base/gamemode/npc.lua:6]: Just call game.GetWorld() at the time the hook is ran - this optimisation is not needed. Also the variable names should be changed to reflect the fact that they're an entity now, not a class string.

**Review by Kefta — CHANGES_REQUESTED** [review]

**Comment by MoskalykA** [comment]
> Have you actually tested this change on an actual server? The reason for sending the classname is that the entity may be deleted by the time the message gets to the client.

I've actually tested my modifications but I hadn't thought of that, I'll test it tomorrow.

**Comment by MoskalykA** [comment]
> I believe net messages will always reach the client before its knowledge of entity deletion when dispatched at the same time, but this should be tested extensively. I'll do some this weekend.

Interesting, I'll let you check it out for yourself.

**Comment by FlorianLeChat** [comment]
Have you tested this in an environment where a server is under heavy network load due to a large number of players and network requests can take a **longer** time to be broadcasted to players? Your suggestion seems to be pretty weak in such cases where bad timing is enough to cause an error.

**Comment by MoskalykA** [comment]
> Have you tested this in an environment where a server is under heavy network load due to a large number of players and network requests can take a **longer** time to be broadcasted to players? Your suggestion seems to be pretty weak in such cases where bad timing is enough to cause an error.

I've just tried to get as many requests as possible and it's still working, but I have no way of testing this with a server full of players and sending more requests than I've been able to get.

**Comment by MoskalykA** [comment]
Hi, I added the modification to a friend's server, you were right

**Pull request closed without merging**